cluster-autoscaler/processors/nodegroupset/balancing_processor.go

```diff
@@ -179,5 +186,64 @@ func (b *BalancingNodeGroupSetProcessor) BalanceScaleUpBetweenGroups(context *co
 	return result, nil
 }
 
+// averageNodegroupAllocatableResources collects the average capacity by nodegroup id,
+// so we can add new nodes to the one with the lowest capacity instead of the one with the lowest node count
+func (b *BalancingNodeGroupSetProcessor) averageNodegroupAllocatableResources(context *context.AutoscalingContext, groups []cloudprovider.NodeGroup, nodes []*apiv1.Node) map[string]int {
```

Comment:
template node would not work since they'd all have the same cpu, we need to inspect what is actually running in the cluster
it would work just fine on scale from 0 since it would scale it evenly (fallback path)

Reply:
Oh, interesting. So how does CA know how many nodes to spin up if the number of cores in actual nodes can vary?

Reply:
it's a gamble just like when spinning up new nodes to an existing cluster 😞 
... basically we get whatever aws has available since we allow multiple different sizes